EventSourcing.Core/Record.cs

```diff
@@ -0,0 +1,62 @@
+namespace EventSourcing.Core;
+
+public record Record
+{
+  /// <summary>
+  /// Unique Partition identifier
+  /// </summary>
+  public Guid PartitionId { get; init; }
+  
+  /// <summary>
+  /// Unique Aggregate identifier
+  /// </summary>
+  public Guid AggregateId { get; init; }
+  
+  /// <summary>
+  /// Unique Record identifier
+  /// </summary>
+  public Guid RecordId { get; init; }
+  
+  /// <summary>
+  /// Index of this Event in the Aggregate Event Stream
+  /// </summary>
+  public long Index { get; init; }
+  
+  /// <summary>
+  /// Aggregate type
+  /// </summary>
+  public string AggregateType { get; init; }
+  
+  /// <summary>
+  /// Record type
+  /// </summary>
+  public string Type { get; init; }
+    
+  /// <summary>
+  /// Record creation time
+  /// </summary>
+  public DateTimeOffset Timestamp { get; init; }
+
+  /// <summary>
+  /// Unique Database Identifier
+  /// </summary>
+  public string id => GetId(AggregateId, Index);
+  
+  /// <summary>
+  /// Create new Event
+  /// </summary>
+  public Record()
+  {
+    RecordId = Guid.NewGuid();
+    Type = GetType().FullName;
```

Comment:
Using 'FullName' could be a problem when rehydrating but the dev has moved an event to a different namespace (Which should be possible IMO)